tests/acceptance/Subscribers/ExportSubscribersCest.php

```diff
@@ -12,7 +12,9 @@ public function exportSubscribers(\AcceptanceTester $i) {
     $segment = $segmentFactory->withName($segmentName)->create();
     $subscriberFactory = new Subscriber();
     $subscriberFactory->withSegments([$segment])->withEmail('one@fake.fake')->create();
+    $subscriberFactory = new Subscriber();
```

Comment:
I think that this doesn't seem right. The factory should work right when it is created only once.
I tried to run the test locally, and the problem is in the method `withSegments` because the function array_merge adds the segment again. After that, the `create` method tries to create an association twice for the same segment.

Reply:
We solved this issue in some newer data factories, by returning an updated clone for every change. See https://github.com/mailpoet/mailpoet/blob/master/tests/DataFactories/WooCommerceCustomer.php and look at update method. It would be nice if we could use it also in other factories so that we don't have to create a new instance of a factory all the time.